example/define_env.yaml

```diff
@@ -0,0 +1,18 @@
+define_env:
+  fields:
+    APP_TYPE:
+      type: String
+      default:
+    BOOL_VALUE:
+      type: bool
+    INT_VALUE:
+      type: int
+    ENUM_VALUE:
+      type: enum
+      enum_values:
+        developer:
+        demo:
+        production:
```

Comment:
enum values would ideally be an array

so I think this should be 

```yaml
    enum_values:
        - developer
        - demo
        - production
```

Reply:
I preferred to use a map because in this way you can also define the name of the enum in the dart class that will be generated. If you have a better idea for doing this I will be happy to implement it
For example:
```yaml
enum_values:
  developer: dev
  demo: demo
  production: prod
```
```yaml
enum_values:
        <ENV_NAME>: <DART_NAME>
```
The risk of a repeated enum value is also avoided

Reply:
I still can't understand how this would work. Could you give me a better example?
from what I can understand the `ENV_NAME` is changed to `DART_NAME` when creating the dart define string? 

I can see this is under FIELD_NAME so would my yaml look like this for APP_ENV

```yaml

define_env:
    fields:
        APP_ENV:
            type: enum
            default: local
            enum_values:
                local: local
                staging: staging
                production: production
```

What happens if we apply the same value for different keys? 

```yaml

define_env:
    fields:
        APP_ENV:
            type: enum
            default: local
            enum_values:
                local: staging
                staging: staging
                production: production
```

Reply:
In the future when you check the settings in yaml it will be checked that there are no duplicates, in case it will issue an error
The resulting class will look like this:
```dart
class Env {
  static const AppEnv appEnv = AppEnv(String.fromEnviroment('APP_ENV', defaultValue: 'local'));
}
class AppEnv {
  final String name;
  const AppEnv(this.name);
  static const AppEnv developer = AppEnv('dev');
  static const AppEnv demo = AppEnv('demo');
  static const AppEnv prod = AppEnv('production');
}
```
Maybe it would be revised, it would be nice to be able to set the name of the dart class as well